aiosmtpd/smtp.py

```diff
@@ -663,9 +885,8 @@ async def smtp_DATA(self, arg):
             return
         # Remove extraneous carriage returns and de-transparency
         # according to RFC 5321, Section 4.5.2.
-        for i in range(len(data)):
-            text = data[i]
-            if text and text[:1] == b'.':
+        for i, text in enumerate(data):
+            if text.startswith(b'.'):
```

Comment:
Slicing seems to be quite slow on PyPy3 (subjective feeling).

This is simpler, clearer, and faster. Three bool-ish ops + 1 slicing replaced by an optimized search.